backend/tracim_backend/applications/upload_permissions/controller.py

```diff
@@ -0,0 +1,164 @@
+    return configurator
+
+
+class UploadPermissionController(Controller):
+    """
+    Endpoints for Upload Permission
+    """
+
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__WORKSPACE_UPLOAD_PERMISSION_ENDPOINTS])
+    @check_right(is_content_manager)
+    @hapic.input_path(WorkspaceIdPathSchema())
+    @hapic.input_body(UploadPermissionCreationBodySchema())
+    @hapic.output_body(UploadPermissionSchema(many=True))
+    def add_upload_permission(
+        self, context, request: TracimRequest, hapic_data=None
+    ) -> typing.List[UploadPermissionInContext]:
+        """
+        Allow to add permission for upload to external person
+        """
+        app_config = request.registry.settings["CFG"]  # type: CFG
+        api = UploadPermissionLib(
+            current_user=request.current_user, session=request.dbsession, config=app_config
+        )
+        upload_permission = api.add_permission_to_workspace(
+            request.current_workspace, hapic_data.body.emails, hapic_data.body.password
+        )
+        return api.get_upload_permissions_in_context(upload_permission)
+
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__WORKSPACE_UPLOAD_PERMISSION_ENDPOINTS])
+    @check_right(is_content_manager)
+    @hapic.input_query(UploadPermissionListQuerySchema())
+    @hapic.input_path(WorkspaceIdPathSchema())
+    @hapic.output_body(UploadPermissionSchema(many=True))
+    def get_upload_permission(
+        self, context, request: TracimRequest, hapic_data=None
+    ) -> typing.List[UploadPermissionInContext]:
+        """
+        Get all upload permission of this workspace
+        """
+        app_config = request.registry.settings["CFG"]  # type: CFG
+        api = UploadPermissionLib(
+            current_user=request.current_user,
+            session=request.dbsession,
+            config=app_config,
+            show_disabled=hapic_data.query.show_disabled,
+        )
+        upload_permission = api.get_upload_permissions(request.current_workspace)
+        return api.get_upload_permissions_in_context(upload_permission)
+
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__WORKSPACE_UPLOAD_PERMISSION_ENDPOINTS])
+    @hapic.handle_exception(UploadPermissionNotFound, HTTPStatus.BAD_REQUEST)
+    @check_right(is_content_manager)
+    @hapic.input_path(UploadPermissionIdPathSchema())
+    @hapic.output_body(NoContentSchema(), default_http_code=HTTPStatus.NO_CONTENT)
+    def disable_upload_permission(self, context, request: TracimRequest, hapic_data=None) -> None:
+        """
+        remove an upload permission
+        """
+        app_config = request.registry.settings["CFG"]  # type: CFG
+        api = UploadPermissionLib(
+            current_user=request.current_user, session=request.dbsession, config=app_config
+        )
+        api.disable_upload_permission(
+            request.current_workspace, hapic_data.path.upload_permission_id
+        )
+        return
```

Comment:
You can remove `return` here. Nothing change but less code is better code ? :p

Reply:
done;